test/modules/TestResultCoordinator/Reports/EdgeHubRestartTest/EdgeHubRestartDirectMethodReportGenerator.cs

```diff
@@ -0,0 +1,303 @@
+
+            long previousSeqNum = 0;
+            ulong passedDirectMethodCount = 0;
+            ulong senderResultCount = 0;
+            ulong receiverResultCount = 0;
+            bool hasSenderResult = true;
+            long senderSeqNum = 0;
+            bool hasReceiverResult = true;
+            long receiverSeqNum = 0;
+
+            (senderResultCount, hasSenderResult, senderSeqNum) =
+                await this.MoveNextSenderResultAsync(senderResultCount);
+
+            (receiverResultCount, hasReceiverResult, receiverSeqNum) =
+                await this.MoveNextReceiverResultAsync(receiverResultCount);
+
+            while (hasSenderResult && hasReceiverResult)
+            {
+                this.ValidateResult(
+                    this.SenderTestResults.Current,
+                    this.SenderSource,
+                    TestOperationResultType.EdgeHubRestartDirectMethod.ToString());
+
+                this.ValidateResult(
+                    this.ReceiverTestResults.Current,
+                    this.ReceiverSource,
+                    TestOperationResultType.DirectMethod.ToString());
+
+                if (receiverSeqNum > senderSeqNum)
+                {
+                    // Increment sender result to have the same seq as the receiver
+                    (senderResultCount, hasSenderResult, senderSeqNum) = await this.IncrementSequenceNumberAsync(
+                        hasSenderResult,
+                        this.MoveNextSenderResultAsync,
+                        receiverSeqNum,
+                        senderResultCount);
+                }
+
+                if (receiverSeqNum < senderSeqNum)
+                {
+                    // Increment receiver result to have the same seq as the sender
+                    (receiverResultCount, hasReceiverResult, receiverSeqNum) = await this.IncrementSequenceNumberAsync(
+                        hasReceiverResult,
+                        this.MoveNextReceiverResultAsync,
+                        senderSeqNum,
+                        receiverResultCount);
+                }
+
+                if (hasSenderResult ^ hasReceiverResult)
+                {
+                    break;
+                }
+
+                // Note: In the current verification, if the receiver obtained more result dm than
+                //   sender, the test is consider a fail. We are disregarding duplicated/unique seqeunce number
+                //   of receiver's result.
+
+                // Check if the current dm is passing
+                bool isCurrentDirectMethodPassing = true;
+                EdgeHubRestartDirectMethodResult senderResult = JsonConvert.DeserializeObject<EdgeHubRestartDirectMethodResult>(this.SenderTestResults.Current.Result);
+
+                // Verified the sequence numbers are the same
+                isCurrentDirectMethodPassing &= senderSeqNum == receiverSeqNum;
```

Comment:
combine these 3 conditions in a single statement.
```suggestion
                isCurrentDirectMethodPassing = (senderSeqNum == receiverSeqNum) && (senderSeqNum > previousSeqNum) && (senderResult.DirectMethodCompletedStatusCode == HttpStatusCode.OK);
```

Reply:
I'd rather have them spread across 3 lines and be very very explicit of what each passing conditions are instead of combining into one massive if condition. I'm not going to address this comment. Sorry.

Reply:
The reason to combine is to avoid set isCurrentDirectMethodPassing to true initially.  You are welcome to use other syntax/approach in order to prevent isCurrentDirectMethodPassing initialize to true at the beginning.  Even if-statement is used, these 3 conditions can still be written in different lines, I think it is still very clear.  Please kindly address.

Reply:
Please address if instead of bitwise operator in both report generator.  thanks.

Reply:
please put in multiple lines

Reply:
TODO: Centralize this conditions for verification.